core/libs/shared/src/core_ops.rs

```diff
@@ -30,23 +30,41 @@ where
     Local: Stagable<F = Base::F>,
 {
     pub fn finalize(&mut self, id: &Uuid, account: &Account) -> SharedResult<File> {
-        let meta = self.find(id)?;
+        let meta = self.find(id)?.clone();
         let file_type = meta.file_type();
         let parent = *meta.parent();
         let last_modified = meta.timestamped_value.timestamp as u64;
         let name = self.name(id, account)?;
         let id = *id;
         let last_modified_by = account.username.clone();
 
-        Ok(File {
-            id,
-            parent,
-            name,
-            file_type,
-            last_modified,
-            last_modified_by,
-            shares: Vec::new(), // todo
-        })
+        // todo: this is a hack to get fuzzer running on sharing
+        let mut shares = Vec::new();
+        for user_access_key in meta.user_access_keys() {
```

Comment:
Can't you just clone here?

Reply:
The hack being right now we're not caching keys, so most shares are going to show up as unknown?

Reply:
I think one of the above statements moves some portion of `meta`, but I forget. Pretty sure impact of cloning here vs there is going to be negligible tho

Reply:
Yeah that is the hack